services/ui-src/src/utils/other/time.ts

```diff
@@ -14,6 +14,24 @@ export const noon: TimeShape = {
   second: 0,
 };
 
+/*
+ * Defines the beginning and end of Period 1 and Period 2.
+ *     Period 1 is from 01/01 to 06/30.
+ *     Period 2 is from 07/01 to 12/31.
+ */
+export const beginningPeriodOne = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 0, 1);
+};
+export const endPeriodOne = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 5, 30);
+};
+export const beginningPeriodTwo = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 6, 1);
+};
+export const endPeriodTwo = (date: string) => {
+  return new Date(new Date(date).getFullYear(), 11, 31);
+};
```

Comment:
periods in comment don't match functions (5/30 vs 6/30 end date for period 1)

Reply:
ooh nice catch, will fix!

Reply:
upon looking at this further, i think my comments do match the function! I can revise the comments to make it more clear. Dates are zero-indexed, so new Date(2023, 5, 30) represents June 30th, 2023